lib/amqp.js

```diff
@@ -114,6 +114,13 @@ class Amqp {
                 + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
         }
 
+        try {
+            await this.publishChannel.waitForConfirms();
+        } catch (error) {
+            log.error('Failed on publishing message to queue!!!');
+            throw new Error('Error failed on publishing to MQ' + error);
```

Comment:
Why do that? Exception will be propagated higher as it can't be handled on this level IMHO.